src/TreeSupport.cpp

```diff
@@ -0,0 +1,250 @@
+//Copyright (c) 2017 Ultimaker B.V.
+//CuraEngine is released under the terms of the AGPLv3 or higher.
+
+#include "utils/intpoint.h" //To normalize vectors.
+#include "utils/math.h" //For round_up_divide and PI.
+#include "utils/MinimumSpanningTree.h" //For connecting the correct nodes together to form an efficient tree.
+#include "utils/polygonUtils.h" //For moveInside.
+
+#include "TreeSupport.h"
+
+#define SQRT_2 1.4142135623730950488 //Square root of 2.
```

Comment:
This can be a `const` and moved to some common header file.

Reply:
It's already a preprocessor macro. It can't be made const.

I'm not really in favour of creating generic common header files because they tend to become a cesspool of stuff since there is no clearly defined scope for these header files. It's not like we need to adjust the value of SQRT_2 that often, so I'd rather keep it close to where it's needed.